### PR TITLE
Fix drbd hook and update to use RPC

### PR DESCRIPTION
**Pull request by MatthewJohn**

### CODE REVIEW

**Comment by MatthewJohn** [comment]
Output:
    root@sol:~# DRBD_RESOURCE=mcvirt_vm-test-vm2-disk-1 python /usr/lib/mcvirt/mcvirt-drbd.py 
    root@sol:~# mcvirt start test-vm2
    Username: mjc
    Password: 
    The last Drbd verification of the Drbd volume failed: mcvirt_vm-test-vm2-disk-1. Run MCVirt as a superuser with --ignore-drbd to ignore this issue

File: source/usr/lib/mcvirt/auth/factory.py [hunk 1]
```diff
@@ -41,10 +42,21 @@ def get_user_types(self):
         return sorted(UserBase.__subclasses__(),
                       key=lambda user_class: user_class.SEARCH_ORDER)
 
+    def get_user_type_by_name(self, user_type):
+        """Return the user_type class for a given name"""
+        for user_type in self.get_user_types():
+            if user_type == user_type.__name__:
+                return user_type
+        raise InvalidUserTypeException('An invalid user type has been passed.')
+
```
Comment (joe-singleton-itdev) [inline source/usr/lib/mcvirt/auth/factory.py]: Is this function used anywhere?

File: source/usr/lib/mcvirt/mcvirt-drbd.py [hunk 2]
```diff
@@ -18,32 +18,31 @@
...
+# Create RPC connection and obtain hard drive factory
+rpc = Connection(username=config['username'], password=config['password'])
+hard_drive_factory = rpc.get_connection('hard_drive_factory')
+
+# Obtain hard drive object and set sync state
+hard_drive_object = hard_drive_factory.getDrbdObjectByResourceName(
+    drbd_resource
+)
+rpc.annotate_object(hard_drive_object)
+hard_drive_object.setSyncState(False, update_remote=False)
```
Comment (joe-singleton-itdev) [inline source/usr/lib/mcvirt/mcvirt-drbd.py]: Should drbd_global.conf have been updated too?

File: source/usr/lib/mcvirt/auth/factory.py [hunk 3]
```diff
@@ -41,10 +42,21 @@ def get_user_types(self):
...
     def ensure_valid_user_type(self, user_type):
         """Ensure that a given user_type is valid."""
-        if user_type not in self.get_user_types():
-            raise InvalidUserTypeException('An invalid user type has been passed')
+        for user_type_itx in self.get_user_types():
+            if (user_type is user_type_itx or
+                    (type(user_type) is str and user_type == user_type_itx.__name__)):
```
Comment (sts-itdev) [inline source/usr/lib/mcvirt/auth/factory.py]: Is "type(user_type) is str" necessary?

File: source/usr/lib/mcvirt/auth/factory.py (hunk 1 above)
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/auth/factory.py]: Nope :D

File: source/usr/lib/mcvirt/mcvirt-drbd.py (hunk 2 above)
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/mcvirt-drbd.py]: I don't think so. The command is still called the same.

File: source/usr/lib/mcvirt/auth/factory.py (hunk 3 above)
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/auth/factory.py]: Yes, this allows either the name of the class or a reference to the class. (Generally references are better) But from the parser/hook script, it doesn't have access to the classes

**Comment by joe-singleton-itdev** [comment]
Ship it! :shipit:

**Pull request merged**